### PR TITLE
fix: missing await in rollback

### LINKED ISSUE #668: Transaction rollback is missing await
https://github.com/googleapis/nodejs-datastore/blob/380c45b3a9566037c3daa37d5b5b85ba318fa281/samples/tasks.markdone.js#L47

Transaction rollback is missing await. According to documentation `transaction.rollback()` returns `Promise`.
 
https://googleapis.dev/nodejs/datastore/latest/Transaction.html#rollback

### PR DESCRIPTION
**Pull request by laljikanjareeya**
Fixes #668

### CODE REVIEW

File: src/request.ts [hunk 1]
```diff
@@ -1321,7 +1325,7 @@ class DatastoreRequest {
         const [response] = await transaction.commit();
         callback!(null, response);
       } catch (err) {
-        transaction.rollback();
+        await transaction.rollback();
```
Comment (AVaksman) [inline src/request.ts:1328]: should this also be wrapped in `try-catch` as above?

File: src/request.ts [hunk 2]
```diff
@@ -1301,7 +1301,11 @@ class DatastoreRequest {
     const transaction = this.datastore.transaction();
     transaction.run(async err => {
       if (err) {
-        transaction.rollback();
+        try {
+          await transaction.rollback();
+        } catch (error) {
+          // rollbacks are fine as best effort.
```
Comment (AVaksman) [inline src/request.ts:1307]: Can probably add reference to a related discussion
```ts
// Even a failed rollback should be transparent.
// RE: https://github.com/GoogleCloudPlatform/gcloud-node/pull/1369#discussion_r66833976
```

**Comment by laljikanjareeya** [comment]
@AVaksman  It shows code coverage decrease in `src/index.ts` but, there is not change on that file.

**Comment by laljikanjareeya** [comment]
@AVaksman Addressed all the comments please take a look!!

**Pull request merged**